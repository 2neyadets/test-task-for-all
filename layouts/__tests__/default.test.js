import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import DefaultLayout from '../default'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('DefaultLayout component', () => {
  let wrapper, actions, getters, store

  beforeEach(() => {
    actions = {
      'notes/getInitialNotesFromStorage': jest.fn(),
      screenChanged: jest.fn(),
    }
    getters = {
      viewportHeight: () => 640,
      viewportWidth: () => 360,
    }
    store = new Vuex.Store({
      actions,
      getters,
    })
    wrapper = shallowMount(DefaultLayout, {
      localVue,
      store,
      stubs: ['Nuxt', 'NuxtLink'],
    })
  })

  it('mounts without errors', done => {
    localVue.config.errorHandler = done // для отлавливания ошибок внутри nextTick, если нужно что-то выпонять внутри него
    expect.assertions(1)
    expect(wrapper).toBeTruthy()
    done()
  })

  it('show/hide task link depends on screen width', () => {
    expect.assertions(2)
    expect(wrapper.findAll('NuxtLink-stub').length).toBe(1)
    const localGetters = {
      viewportHeight: () => '100vh',
      viewportWidth: () => '100vw',
    }
    const localStore = new Vuex.Store({
      actions,
      getters: localGetters,
    })
    const localWrapper = shallowMount(DefaultLayout, {
      localVue,
      store: localStore,
      stubs: ['Nuxt', 'NuxtLink'],
    })
    expect(localWrapper.findAll('NuxtLink-stub').length).toBe(2)
  })

  it('app height === viewportHeight', () => {
    expect.assertions(1)
    expect(wrapper.find('.column').attributes('style')).toBe(`height: ${wrapper.vm.viewportHeightLayout};`)
  })

  it('getInitialNotesFromStorage was called', () => {
    expect.assertions(1)
    expect(actions['notes/getInitialNotesFromStorage']).toHaveBeenCalled()
  })

  it('interact with windowResized', () => {
    expect.assertions(2)
    const startLenghtOfCalls = actions.screenChanged.mock.calls.length
    wrapper.vm.windowResized(undefined)
    expect(actions.screenChanged.mock.calls.length).toBe(startLenghtOfCalls)
    wrapper.vm.windowResized({ innerWidth: 1024, innerHeight: 768 })
    expect(actions.screenChanged.mock.calls.length).not.toBe(startLenghtOfCalls)
  })

  it('match snapshot', () => {
    expect.assertions(1)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('remove event listener before destroy wrapper', () => {
    wrapper.destroy()
  })
})
