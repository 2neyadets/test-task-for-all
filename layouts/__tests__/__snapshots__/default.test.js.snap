// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout component match snapshot 1`] = `
<div
  class="text-white"
>
  <div
    class="column"
    style="height: 640px;"
  >
    <div
      class="col-auto"
    >
      <header
        class="header bg-grey-8"
      >
        <div
          class="toolbar row justify-between items-center no-wrap"
        >
          <nuxtlink-stub
            class="text-white"
            to="/"
          >
            test-task-for-all
          </nuxtlink-stub>
          <!---->
          <a
            class="text-white"
            href="mailto:avsintsov91@gmail.com"
          >
            <i
              class="material-icons"
            >
              mail
            </i>
            <span
              class="px-sm"
            >
              avsintsov91@gmail.com
            </span>
          </a>
        </div>
        <div
          class="layout__shadow absolute-full overflow-hidden no-pointer-events"
        />
      </header>
    </div>
    <div
      class="col"
    >
      <div
        class="main full-height overflow-auto bg-dark"
      >
        <nuxt-stub />
      </div>
    </div>
  </div>
  <deletenotedialog-stub />
</div>
`;
